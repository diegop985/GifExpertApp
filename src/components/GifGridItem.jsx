import React from 'react';
import 'animate.css';
import PropTypes from 'prop-types';

export const GifGridItem = ({ id, title, img_url }) => {


  return (
    <div className='card animate__animated animate__bounceIn'>
        <img src={ img_url } alt= { title }/>
        <p> {title} </p>
    </div>
  )
}


GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  img_url: PropTypes.string.isRequired
}

