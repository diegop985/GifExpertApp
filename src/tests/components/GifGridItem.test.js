import { render } from "@testing-library/react"
import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import { GifGridItem } from "../../components/GifGridItem";


describe('Tests on <GifGridItem/>', () => { 

    const title = "Un titulo"
    const url = "https://localhost/thisreactapp.jsx"
    
    const wrapper = shallow(
        <GifGridItem
            title={title}
            img_url={url}
        />
    )
    
    test('should render <GifGridItem/> correctly', () => { 
        
        
        expect( wrapper ).toMatchSnapshot()


    })

    test('should check for proptypes', () => {

        expect( wrapper ).toMatchSnapshot()


    })

    test('should have a paragraph with the title', () => { 

        const p = wrapper.find('p')

        expect( p.text().trim()).toBe( title )

    })

    test('should obtain an img with url and alt from props', () => { 

        const img = wrapper.find('img')

        expect( img.prop('src') ).toBe(url)
        expect( img.prop('alt') ).toBe(title)

    })

    test('should have "animate__bounceIn" className', () => { 
        

        const div = wrapper.find('div')

        expect( div.hasClass('animate__bounceIn')).toBe(true)
    })
})


