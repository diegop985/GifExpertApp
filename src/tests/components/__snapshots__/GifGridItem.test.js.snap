// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem/> should check for proptypes 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/thisreactapp.jsx"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;

exports[`Tests on <GifGridItem/> should render <GifGridItem/> correctly 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/thisreactapp.jsx"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
